### ISSUE TITLE
The use of the "mask"

### ISSUE THREAD

**Issue by zhengmingzhang** [narrowsnap/pytorch_STN#2]
`
class STN(nn.Module):

    def __init__(self):
        super(STN, self).__init__()
        self.conv = conv1x1(512, 128)   
        self.fc_loc = nn.Sequential(
            nn.Linear(128*14*14, 64),
            nn.Tanh(),
            nn.Linear(64, 6),
            nn.Tanh()
        )

    def forward(self, feature):
        feature = self.conv(feature)
        feature = feature.view(-1, 128*14*14)
        theta = self.fc_loc(feature)

        mask = torch.tensor([[1, 0, 0, 0, 0, 0], 
                             [0, 0, 0, 0, 0, 0], 
                             [0, 0, 1, 0, 0, 0],
                             [0, 0, 0, 0, 0, 0],
                             [0, 0, 0, 0, 1, 0],
                             [0, 0, 0, 0, 0, 1]], dtype=torch.float).cuda()
        theta = torch.mm(theta, mask)
        theta = theta.view(-1, 2, 3)
        return theta`
May i know what's the use of the mask?thank you!

**Comment by narrowsnap** [comment]
Sorry for the late reply. The original theta has 6 parameters, assume is [[a, b, c], [d, e, f]]. The parameters of b and d control the shear of the image. I don't want to shear the feature map. So I use a mask to make them be zero. The shear parameters may help in the pedestrian recognition problems.

**Comment by zhengmingzhang** [comment]
> Sorry for the late reply. The original theta has 6 parameters, assume is [[a, b, c], [d, e, f]]. The parameters of b and d control the shear of the image. I don't want to shear the feature map. So I use a mask to make them be zero. The shear parameters may help in the pedestrian recognition problems.

oh~~~thanks!
